# nf-wrap/main: Output

## Introduction

This document describes the output produced by the tools.

The directories listed below will be created in the results directory once finished.
All paths are relative to the top-level results directory.

### FastQC

<details markdown="1">
<summary>Output files</summary>

- `fastqc/`
  - `*_fastqc.html`: FastQC report containing quality metrics.
  - `*_fastqc.zip`: Zip archive containing the FastQC report, tab-delimited data file and plot images.

</details>

[FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/) gives general quality metrics about your sequenced reads. It provides information about the quality score distribution across your reads, per base sequence content (%A/T/G/C), adapter contamination and overrepresented sequences. For further reading and documentation see the [FastQC help pages](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/).

### General information

<details markdown="1">
<summary>Output files</summary>

- `general_info/`
  - Reports generated by Nextflow: `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `general_dag.dot`/`general_dag.svg`.
  - Reports generated by the project: `general_report.html`, `general_report.txt` and `software_versions.yml`. The `general_report*` files will only be present if the `--email` / `--email_on_fail` parameter's are used when running the project.

</details>

[Nextflow](https://www.nextflow.io/docs/latest/tracing.html) provides excellent functionality for generating various reports relevant to the running and execution of the general. This will allow you to troubleshoot errors with the running of the general, and also provide you with other information such as launch commands, run times and resource usage.
